runtime/array.c

```diff
@@ -181,6 +270,22 @@ CAMLprim value caml_array_unsafe_set(value array, value index, value newval)
   return caml_array_unsafe_set_addr(array, index, newval);
 }
 
+/* [ local_ 'a array -> int -> local_ 'a -> unit ]
+
+   Must be used carefully, as it can violate the "no forward pointers"
+   restriction on the local stack. */
```

Comment:
Should the setting functions above with the bounds checks also have this comment?